shell/platform/android/io/flutter/embedding/android/KeyEmbedderResponder.java

```diff
@@ -0,0 +1,365 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.view.KeyEvent;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.flutter.embedding.android.KeyboardMap.PressingGoal;
+import io.flutter.embedding.android.KeyboardMap.TogglingGoal;
+import io.flutter.plugin.common.BinaryMessenger;
+import java.util.HashMap;
+
+/**
+ * A {@link KeyboardManager.Responder} of {@link KeyboardManager} that handles events by sending
+ * processed information in {@link KeyData}.
+ *
+ * <p>This class corresponds to the HardwareKeyboard API in the framework.
+ */
+public class KeyEmbedderResponder implements KeyboardManager.Responder {
+  private static final String TAG = "KeyEmbedderResponder";
+
+  // Maps KeyEvent's action and repeatCount to a KeyData type.
+  private static KeyData.Type getEventType(KeyEvent event) {
+    final boolean isRepeatEvent = event.getRepeatCount() > 0;
+    switch (event.getAction()) {
+      case KeyEvent.ACTION_DOWN:
+        return isRepeatEvent ? KeyData.Type.kRepeat : KeyData.Type.kDown;
+      case KeyEvent.ACTION_UP:
+        return KeyData.Type.kUp;
+      default:
+        throw new AssertionError("Unexpected event type");
+    }
+  }
+
+  // The messenger used to send Flutter key events to the framework.
```

Comment:
```suggestion
  // The messenger that is used to send Flutter key events to the framework.
```

When I first read it, I thought "why is he talking about how it worked in the past?" :-) It makes sense as is, but is ambiguous.